packages/squiggle-lang/__tests__/Distributions/AlgebraicCombination_test.res

```diff
@@ -0,0 +1,313 @@
+                    | Some(y) => x -> expect -> toBeSoCloseTo(y, ~digits=0)
+                }
+            }
+        })
+        test("(normal(mean=10) + normal(mean=10)).pdf(1.9e1)", () => {
+            let received = normalDist20
+                -> Ok
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, 1.9e1))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toOption
+                -> E.O.flatten
+            let calculated = normalDist10
+                -> algebraicAdd(normalDist10)
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, 1.9e1))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toOption
+                -> E.O.flatten
+            switch received {  
+                | None => "this branch occurs when the dispatch to Jstat on trusted input fails." -> expect -> toBe("never")
+                | Some(x) => switch calculated {
+                    | None => "algebraicAdd has" -> expect -> toBe("failed")
+                    | Some(y) => x -> expect -> toBeSoCloseTo(y, ~digits=1)
+                }
+            }
+        })
+        test("(uniform(low=9, high=10) + beta(alpha=2, beta=5)).pdf(10)", () => {
+            let received = uniformDist 
+                -> algebraicAdd(betaDist)
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, 1e1))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=4. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.001978994877226945, ~digits=3)  
+            }
+        })
+        test("(beta(alpha=2, beta=5) + uniform(low=9, high=10)).pdf(10)", () => {
+            let received = betaDist 
+                -> algebraicAdd(uniformDist)
+                -> E.R2.fmap(d => GenericDist_Types.Constructors.UsingDists.pdf(d, 1e1))
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
+                // sometimes it works with ~digits=4. 
+                | Some(x) => x -> expect -> toBeSoCloseTo(0.001978994877226945, ~digits=3) 
+            }
+        })
+    })
+    describe("cdf", () => {
+        testAll("(normal(mean=5) + normal(mean=5)).cdf (imprecise)", list{6e0, 8e0, 1e1, 1.2e1}, x => {
```

Comment:
this really isn't clear what it's testing. There should be some better way of doing this. Maybe something like, each test turns into the string,
```(normal(mean=5)+normal(mean(5))).cdf(`${6e0}`) should equal normal(mean=10).cdf(`${6e0}`)```